### ISSUE TITLE
App crash(the fga button suddenly disappears during quest) also when reopening, after the crash, also crashes for a while before opening

### ISSUE THREAD

**Issue by ArthurKun21** [Fate-Grand-Automata/FGA#1501]
### Preparation

- [X] I tested the [latest release](https://github.com/Fate-Grand-Automata/FGA/releases/tag/canary)
- [X] I looked at other issues (even the [closed ones](https://github.com/Fate-Grand-Automata/FGA/issues?q=is%3Aissue+is%3Aclosed))
- [X] I read the [Troubleshooting Guide](https://github.com/MathewSachin/Fate-Grand-Automata/wiki/Troubleshooting)

### FGO server

NA

### FGA build number

1908

### Describe the bug

1. Start the app
2. Start FGO
3. Go to the quest screen
4. Start the FGA
5. wait until the wave 1
6. suddenly the fga crashes(the fga button suddenly gone)

sometimes the app just suddenly crashes at wave 1 or while loading into wave 1

crashing when opening fga log (this is on the first video)
[crashing when opening fga log.txt](https://github.com/Fate-Grand-Automata/FGA/files/10275897/crashing.when.opening.fga.log.txt)

fga gone suddenly log (this is on the second video)
[fga gone suddenly log.txt](https://github.com/Fate-Grand-Automata/FGA/files/10275899/fga.gone.suddenly.log.txt)

Other log files that might help, also captured during their own crashes but no video of their crashes

[fga log.txt](https://github.com/Fate-Grand-Automata/FGA/files/10275921/fga.log.txt)
[fga log2.txt](https://github.com/Fate-Grand-Automata/FGA/files/10275922/log2.txt)

### Video

The fga button suddenly gone
https://youtu.be/wXKNEqyLgsY

When opening will crash
https://www.youtube.com/watch?v=Zs2pif9WLQs

### Device model

Xiaomi Redmi note 7 (Pixel Experience rom)

### Android version

11

### Screen size

2336x1080

### RAM

3GB

**Comment by ArthurKun021** [comment]
one of the crashes
`java.io.FileNotFoundException: No resource found for: android.resource://com.mathewsachin.fategrandautomata.debug/0`

But I even did again the `Extract Default support images` just to be sure. but it still crashes

**Comment by ArthurKun021** [comment]
For future documentation
Assuming you already connected the devices to the adb via `adb devices`

this is the adb command I use to get the crash
```log
adb logcat -e com.mathewsachin.fategrandautomata > log.txt
```

this is for the recording of the file
```
scrcpy -r file.mp4
```

**Comment by Ponpokorinn** [comment]
Thought I was the only one. One thing I did to make it not crash is to just set the game to 30fps instead of 60. It will still crash but after like an hour of use. After that, I need to reset my phone because fga is bugged now and won't start again.

**Comment by ArthurKun021** [comment]
> Thought I was the only one. One thing I did to make it not crash is to just set the game to 30fps instead of 60. It will still crash but after like an hour of use. After that, I need to reset my phone because fga is bugged now and won't start again.

yea it used to be like that for me. But nowadays, the crashing is more frequent

**Comment by reconman** [comment]
@ArthurKun021 Can you share the log? I want to check which part fails.

**Comment by Ponpokorinn** [comment]
Forgot to say that I have Xiaomi Redmi Note 7 too but with stock rom and android 10.

**Comment by reconman** [comment]
"Accessibility service not running" is the cause of the crashes.

Is battery optimization disabled? Is the app "locked" in the "recently opened apps" screen? (Square button in the navigation bar)

If yes, is there any error before the first "Accessibility service is not running" message?

**Comment by ArthurKun021** [comment]
![image](https://user-images.githubusercontent.com/16458204/208961537-d62a4948-7201-404a-bcfd-a3fef258f63f.png)

![image](https://user-images.githubusercontent.com/16458204/208962010-0c02c779-9309-4781-9c58-e45d3353b213.png)

![image](https://user-images.githubusercontent.com/16458204/208962649-103b7b88-9586-40d1-80af-244e866b6895.png)

![image](https://user-images.githubusercontent.com/16458204/208963003-8b0edec3-d2b4-4a11-9342-b80f6adbc78c.png)


> Is battery optimization disabled?

I already have my battery optimization for the FGA turn off

> Is the app "locked" in the "recently opened apps" screen? (Square button in the navigation bar)

On my phone, it is not `pause app`

> If yes, is there any error before the first "Accessibility service is not running" message?

Based from the suggestion here, I did turn it off and then turn it on and then the accessibility is now on again. After the crash upon going to the app again you'll see the state of the accessibility service the same as the 3rd picture

**Comment by reconman** [comment]
Please check if this build works better: https://github.com/Fate-Grand-Automata/FGA/suites/9999667043/artifacts/485474291

FGA shouldn't crash repeatedly anymore, but it will still crash from time to time when the Accessibility Service was killed by Android.

**Comment by ArthurKun021** [comment]
> Please check if this build works better: https://github.com/Fate-Grand-Automata/FGA/suites/9999667043/artifacts/485474291
> 
> FGA shouldn't crash repeatedly anymore, but it will still crash from time to time when the Accessibility Service was killed by Android.

[log.txt](https://github.com/Fate-Grand-Automata/FGA/files/10281723/log.txt)

**Comment by ArthurKun021** [comment]
Good news and bad news I guess.

Bad news. 
I can't seem to find a fix for it. I check some post and comments from stackoverflow and reddit and it seems like the Tasker app, which similarly uses the accessibility settings, faces the same issue. I forgot where I saw it but it seems like this is related to bug in Android. What happened in the Tasker is that they uses a webiew as dependency and when it is updated causes the bug.

I don't know if this links might help but here

From the official android platform which states the problem with the accessibility services
<https://cs.android.com/android/_/android/platform/frameworks/base/+/ec8cff62610d0964b015e7e0918ba060a29bb617>

Relevant links to this
https://www.reddit.com/r/tasker/comments/syhzr7/comment/hxyc37o

https://www.reddit.com/r/tasker/comments/sz2kg7/investigation_accessibility_services_getting/



Good news.
I updated to Android 12 and facing no problem so far but that might be also due to writing additional adb commands.

But this might be also due to freshly installed android 12 hence no issues so far.

I will still keep checking on it due to the lotto ongoing right now ~~I need to catch up~~

You might want to run some adb commands

For disabling the battery optimization to the specific app. just change `-` to `+` if you want to re-enable
```shell
adb shell dumpsys deviceidle whitelist -=com.mathewsachin.fategrandautomata
```

To help with the accessibility settings
```shell
adb shell settings put secure enabled_accessibility_services com.mathewsachin.fategrandautomata/com.mathewsachin.fategrandautomata
```
